cmake/CubBuildCompilerTargets.cmake

```diff
@@ -38,6 +39,7 @@ function(cub_build_compiler_targets)
     # Some tests require /bigobj to fit everything into their object files:
     append_option_if_available("/bigobj" cxx_compile_options)
   else()
+    append_option_if_available("-Wreorder" cuda_compile_options)
```

Comment:
Could we please separate that one from the others.

It is really easy to overlook that one is `cuda_compile_options` and the others `cxx_compile_options`